devito/ir/clusters/cluster.py

```diff
@@ -42,12 +43,18 @@ def __init__(self, exprs, ispace=None, guards=None, properties=None, syncs=None)
 
         self._exprs = tuple(ClusterizedEq(e, ispace=ispace) for e in as_tuple(exprs))
         self._ispace = ispace
-        self._guards = frozendict(guards or {})
+        self._guards = Guards(guards or {})
```

Comment:
note for reviewers: to be exploited inside the compiler incrementally, so far only in a few places (and mostly in PRO)